velox/functions/prestosql/geospatial/GeometrySerde.h

```diff
@@ -17,17 +17,346 @@
 #pragma once
 
 #include <geos/geom/Geometry.h>
+#include <geos/geom/GeometryFactory.h>
+
+#include "velox/common/base/IOUtils.h"
+#include "velox/functions/prestosql/geospatial/GeometryUtils.h"
 #include "velox/type/StringView.h"
 
 namespace facebook::velox::functions::geospatial {
 
-/// Deserialize Velox's internal format to a geometry.  Do not call this within
-/// GEOS_TRY macro: it will catch the exceptions that need to bubble up.
-std::unique_ptr<geos::geom::Geometry> deserializeGeometry(
-    const StringView& geometryString);
+enum class GeometrySerializationType : uint8_t {
+  POINT = 0,
+  MULTI_POINT = 1,
+  LINE_STRING = 2,
+  MULTI_LINE_STRING = 3,
+  POLYGON = 4,
+  MULTI_POLYGON = 5,
+  GEOMETRY_COLLECTION = 6,
+  ENVELOPE = 7
+};
+
+enum class EsriShapeType : int {
+  POINT = 1,
+  POLYLINE = 3,
+  POLYGON = 5,
+  MULTI_POINT = 8
+};
+
+/**
+ * VarbinaryWriter is a utility for serializing raw binary data to a
+ * generic writer interface. It supports writing either raw byte arrays
+ * or trivially copyable types.
+ *
+ * @tparam StringWriter A type that provides an `append(std::string_view)`
+ * method, used to consume the binary output. Examples include `std::string` or
+ *         `core::StringWriter`.
+ */
+template <typename StringWriter>
+class VarbinaryWriter {
+ public:
+  VarbinaryWriter(StringWriter& stringWriter) : stringWriter_(stringWriter) {}
+  VarbinaryWriter() = delete;
+
+  void write(const char* data, size_t size) {
+    stringWriter_.append(std::string_view(data, size));
+  }
+
+  template <typename T>
+  void write(const T& value) {
+    VELOX_CHECK(
```

Comment:
since this only depends on the template param, can we make this a static assert instead?, like:

```
static_assert(std::is_trivially_copyable_v<T>, "T must be trivially copyable");
```